src/function/FunctionLink.ts

```diff
@@ -298,11 +306,179 @@ export function addLinkTools(
           .inScheme
       )
-      ].readOnly;
+    ].readOnly;
   let tools = [verticesTool, segmentsTool];
   if (!readOnly) tools.push(removeButton);
   let toolsView = new joint.dia.ToolsView({
-    tools: tools
+    tools: tools,
   });
   linkView.addTools(toolsView);
 }
+
+export async function getCacheConnections(iri: string) {
+  const terms: { [key: string]: any } = {};
+  const restrictions: { [key: string]: Restriction[] } = {};
+  const subClass: string[] = await fetchSubClasses(
+    AppSettings.contextEndpoint,
+    iri
+  );
+  const subClasses = subClass
+    .filter((subClass) => !(subClass in WorkspaceTerms))
+    .concat(
+      WorkspaceTerms[iri].subClassOf.filter(
+        (subClass) => !(subClass in WorkspaceTerms)
+      )
+    );
+  const relationships: {
+    relation: string;
+    target: string;
+    labels: { [key: string]: string };
+  }[] = [];
+  const connections: {
+    link: string;
+    linkLabels: { [key: string]: string };
+    target: {
+      iri: string;
+      labels: { [key: string]: string };
+      description: { [key: string]: string };
+      vocabulary: string;
+    };
+    direction: string;
+  }[] = [];
+
+  if (AppSettings.representation === Representation.FULL) {
+    await fetchRelationships(AppSettings.contextEndpoint, [iri], restrictions);
+    await fetchReadOnlyTerms(
+      AppSettings.contextEndpoint,
+      Object.keys(restrictions)
+        .filter((restriction) => !(restriction in WorkspaceTerms))
+        .concat(subClasses)
+        .concat(
+          WorkspaceTerms[iri].restrictions
+            .filter(
+              (restr) =>
+                isUrl(restr.target) && !(restr.target in WorkspaceTerms)
+            )
+            .map((restr) => restr.target)
+        ),
+      terms
+    );
+    for (const restriction of WorkspaceTerms[iri].restrictions) {
+      if (restriction.target in terms) {
+        const newConnection = {
+          link: restriction.onProperty,
+          linkLabels: Links[restriction.onProperty].labels,
+          target: {
+            iri: restriction.target,
+            labels: terms[restriction.target].labels,
+            description: terms[restriction.target].definitions,
+            vocabulary: getVocabularyFromScheme(
+              terms[restriction.target].inScheme
+            ),
+          },
+          direction: "source",
+        };
+        if (
+          !connections.find(
+            (conn) => JSON.stringify(conn) === JSON.stringify(newConnection)
+          )
+        )
+          connections.push(newConnection);
+      }
+    }
+  } else if (AppSettings.representation === Representation.COMPACT) {
+    await fetchFullRelationships(
+      AppSettings.contextEndpoint,
+      iri,
+      relationships
+    );
+    await fetchReadOnlyTerms(
+      AppSettings.contextEndpoint,
+      relationships.map((rel) => rel.target).concat(subClasses),
+      terms
+    );
+    for (const relationship of relationships.filter(
```

Comment:
This is very hard to read. The filtered relationships should be extracted to a variable before the for loop code. Similar pattern occurs on many places in this function.